qadence_protocols/mitigations/twirl.py

```diff
@@ -55,6 +56,14 @@ def mitigate(
             ]
         )
     )
+
+    # Random sample gate strings for scalability
+    twirl_samples = options.get("twirl_samples")
+    if isinstance(twirl_samples, int) and twirl_samples > 0:
+        twirls = random.sample(twirls, twirl_samples)
```

Comment:
It would be clever to not sample from the whole twirl space, but rather generate as much needed, as that would be scalable. Not sure if this is straightforward here, feel free to ignore